src/app/search/search-results/search-results.component.ts

```diff
@@ -42,16 +43,23 @@ export class SearchResultsComponent implements OnInit {
     overflow: false
   };
   constructor(private searchService: SearchService, private queryBuilderService: QueryBuilderService, private searchQuery: SearchQuery) {
+    super();
     this.activeToolTab$ = this.searchQuery.activeToolTab$;
     this.noWorkflowHits$ = this.searchQuery.noWorkflowHits$;
     this.noToolHits$ = this.searchQuery.noToolHits$;
     this.showToolTagCloud$ = this.searchQuery.showToolTagCloud$;
     this.showWorkflowTagCloud$ = this.searchQuery.showWorkflowTagCloud$;
   }
 
+  private readonly TOOLS_TAB_INDEX = 0;
+  private readonly WORKFLOWS_TAB_INDEX = 1;
+
   ngOnInit() {
     this.createTagCloud('tool');
     this.createTagCloud('workflow');
+    this.activeToolTab$
+      .pipe(takeUntil(this.ngUnsubscribe))
+      .subscribe(toolsActive => (this.selectedIndex = toolsActive ? this.TOOLS_TAB_INDEX : this.WORKFLOWS_TAB_INDEX));
```

Comment:
Trying making selectedIndex an observable and pipe mapping activeToolTab to it.  This should let you get rid of the extends Base unsubscribe hassle.

Reply:
The problem we were running into was this HTML snippet: `[(selectedIndex)]="selectedIndex"` -- and Angular doesn't let you specify an observable nor use async there.

Can you clarify how to get around that?

Reply:
`[(selectedIndex)]` => `[seletedIndex]`

Reply:
Made selectedIndex an observable, then pipe mapped activeToolTab to it.